### DISCUSSION TITLE
Sugestão: Excluir Produto

### DISCUSSION THREAD

**Discussion by julianoaraujonk** [RamonSilva20/mapos#1706] [Ideas]
Testei o sistema e achei que poderia mudar a regra para excluir um produto, não permitir a exclusão de um produto caso o mesmo ja tenha sido lançado em alguma venda, ao invés de excluir esse produto seria bom colocar a opção de desativar , avisando que o mesmo ja tem movimento em vendas.

**Comment by willph** [comment]
> Testei o sistema e achei que poderia mudar a regra para excluir um produto, não permitir a exclusão de um produto caso o mesmo ja tenha sido lançado em alguma venda, ao invés de excluir esse produto seria bom colocar a opção de desativar , avisando que o mesmo ja tem movimento em vendas.

Você quer dizer se já for faturado?

**Reply by julianoaraujonk** [reply]
isso, a venda mesmo já faturada o sistema permite a exclusão do produto, removendo o produto tbm da venda que já tinha ele.

**Comment by willph** [comment]
Seria isso no caso?

![teste-delete-produt](https://user-images.githubusercontent.com/17226802/158037279-e849731a-5c63-4eb7-be27-e317e003ad4c.PNG)

**Reply by julianoaraujonk** [reply]
Isso , assim não corre o risco de apagar um produto errado e alterar as vendas

**Reply by julianoaraujonk** [reply]
bloquear o produto tbm na parte de produtos , nao permitir excluir um produto caso ele já tenha sido usado em alguma venda , vi agora que essa tela que vc mandou é na parte de vendas , se for no produto vc consegue apagar o produto que foi lançado ai na nota.

**Reply by julianoaraujonk** [reply]
se for em produtos e apagar o HD 500GB Witech , ele vai ser removido da venda tbm

**Reply by ChrisFrs** [reply]
> bloquear o produto tbm na parte de produtos , nao permitir excluir um produto caso ele já tenha sido usado em alguma venda , vi agora que essa tela que vc mandou é na parte de vendas , se for no produto vc consegue apagar o produto que foi lançado ai na nota.

Provavelmente eles colocaram já uma trava, por que não é possível fazer isso já que se excluir o produto depois de faturado ele quebra o banco de dados, o que da para fazer é adicionar um campo a mais no campo Produtos no Sql, onde se caso ele tiver disponível seja 1 e se não tiver seja 0, OUU manter o produto lá no banco porem desativa-lo e copiar foreign key, mas só para manter sabendo o produto que é mas só isso.
Mas ai no caso já tem também quando o item fica disponível e não disponível devido ao valor inicial ao estoque mínimo.

**Reply by julianoaraujonk** [reply]
Estou com o sistema atualizado e fiz o teste ele apaga o produto e remove da venda , mesma a venda já estando faturada

**Reply by willph** [reply]
Em configurações>financeiro você pode permitir ou bloquear isso
